rpc/dialer.go

```diff
@@ -240,6 +241,145 @@ func DialDirectGRPC(ctx context.Context, address string, logger utils.ZapCompati
 	return dialInner(ctx, address, logger, dOpts)
 }
 
+func socksProxyDialContext(ctx context.Context, network, proxyAddr, addr string) (net.Conn, error) {
+	dialer, err := proxy.SOCKS5(network, proxyAddr, nil, proxy.Direct)
+	if err != nil {
+		return nil, fmt.Errorf("error creating SOCKS proxy dialer to address %q from environment: %w",
+			proxyAddr, err)
+	}
+	conn, err := dialer.(proxy.ContextDialer).DialContext(ctx, network, addr)
```

Comment:
Why do you have to cast to `proxy.ContextDialer` here? Is `DialContext` not available on the SOCKS5 dialer?